elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -73,6 +79,216 @@ describe('peerPool', () => {
+			(peerPool as any)._peerPoolConfig.latencyProtectionRatio = DEFAULT_PEER_PROTECTION_FOR_LATENCY;
+			(peerPool as any)._peerPoolConfig.productivityProtectionRatio = DEFAULT_PEER_PROTECTION_FOR_USEFULNESS;
+			(peerPool as any)._peerPoolConfig.longevityProtectionRatio = DEFAULT_PEER_PROTECTION_FOR_LONGEVITY;
+			sandbox
+				.stub(utils, 'constructPeerIdFromPeerInfo')
+				.returns('notAWhitelistedId');
+		});
+
+		describe('when node using default protection ratio values has 100 inbound peers', () => {
+			beforeEach(() => {
+				sandbox.stub(peerPool, 'getPeers').returns(originalPeers as Peer[]);
+			});
+
+			it('should return expected amount of eviction candidates', async () => {
+				const selectedPeersForEviction = (peerPool as any)._selectPeersForEviction();
+
+				expect(selectedPeersForEviction.length).to.eql(42);
+			});
+		});
+
+		describe('when node using default protection ratio values has 10 inbound peers', () => {
+			beforeEach(() => {
+				originalPeers = [...new Array(10).keys()].map(i => ({
+					netgroup: i,
+					latency: i,
+					responseRate: i % 2 ? 0 : 1,
+					connectTime: i,
+				}));
+				sandbox.stub(peerPool, 'getPeers').returns(originalPeers as Peer[]);
+			});
+
+			it('should return expected amount of eviction candidates', async () => {
+				const selectedPeersForEviction = (peerPool as any)._selectPeersForEviction();
+
+				expect(selectedPeersForEviction.length).to.eql(4);
+			});
+		});
+
+		describe('when node using default protection ratio values has 5 inbound peers', () => {
+			beforeEach(() => {
+				originalPeers = [...new Array(5).keys()].map(i => ({
+					netgroup: i,
+					latency: i,
+					responseRate: i % 2 ? 0 : 1,
+					connectTime: i,
+				}));
+				sandbox.stub(peerPool, 'getPeers').returns(originalPeers as Peer[]);
+			});
+
+			it('should return expected amount of eviction candidates', async () => {
+				const selectedPeersForEviction = (peerPool as any)._selectPeersForEviction();
+
+				expect(selectedPeersForEviction.length).to.eql(1);
+			});
+		});
+
+		describe('when node using default protection ratio values has 0 inbound peers', () => {
+			beforeEach(() => {
+				originalPeers = [];
+				sandbox.stub(peerPool, 'getPeers').returns(originalPeers as Peer[]);
+			});
+
+			it('should return expected amount of eviction candidates', async () => {
+				const selectedPeersForEviction = (peerPool as any)._selectPeersForEviction();
+
+				expect(selectedPeersForEviction.length).to.eql(0);
+			});
+		});
+
+		describe('when node with netgroup protection disabled has 100 inbound peers', () => {
+			beforeEach(() => {
+				(peerPool as any)._peerPoolConfig.netgroupProtectionRatio = 0;
+				sandbox.stub(peerPool, 'getPeers').returns(originalPeers as Peer[]);
+			});
+
+			it('should return expected amount of eviction candidates', async () => {
+				const selectedPeersForEviction = (peerPool as any)._selectPeersForEviction();
+
+				expect(selectedPeersForEviction.length).to.eql(44);
```

Comment:
how we are expecting that there will be 44 peers? I see in most of the places we are using a specific number and we don't know why we are expecting that number. If it's possible can we break it down like `peersWithPoorLatency` and we can use `expect(selectedPeersForEviction.length).to.eql(peersWithPoorLatency.length)`

Reply:
If we compare to a `peersWithPoorLatency`, how do we define the expected threshold for `poorLatency`? The specific numbers can be calculated using the `DEFAULT` values. For example for this test, we have `netgroup` protection disabled. Using default protection ratios, the breakdown for 100 peers is: 4 by `netgroup`, 7 by `latency`, 7 by `usefulness `and 41 by `longevity` (simply multiply the ratio with 100). Since `netgroup` is disabled, that leaves `7 (latency) + 7(usefulness) + 41 (longevity) = 55. 100 - 55 = 45.` The 1 difference is a rounding difference, I opted to round down.

Reply:
I mean if it's possible then can we actually make few variables (`noOfPeersbyNetgroup`, `noOfPeersbyLatency`, .etc) and then sum them up before using it in expect? Maybe just for a test so that its clear what's going on if I'm new and I read the tests directly.

Reply:
It's a good point. I will add a comment detailing the above calculation before the `expect` to make it clear.